src/material-experimental/mdc-input/input.ts

```diff
@@ -33,8 +33,10 @@ import {MatInput as BaseMatInput} from '@angular/material/input';
     '[required]': 'required',
     '[attr.placeholder]': 'placeholder',
     '[attr.readonly]': 'readonly && !_isNativeSelect || null',
-    '[attr.aria-invalid]': 'errorState',
-    '[attr.aria-required]': 'required.toString()',
+    // Only mark the input as invalid for assistive technology if it has a value since the
+    // state usually overlaps with `aria-required` when the input is empty and can be redundant.
+    '[attr.aria-invalid]': 'errorState && !empty',
```

Comment:
Just to clarify- will `errorState` be `true` when the control is touched (not pristine)?

Reply:
The `errorState` is set based on the `ErrorStateMatcher`. The default one will be true if the input is invalid and touched or the form has been submitted.